routes/data.js

```diff
@@ -1,8 +1,73 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express');
+const router = express.Router();
+const firebase = require('firebase');
+
+if (!firebase.apps.length) {
+	firebase.initializeApp(config);
+}
+
+// Creates connection to database.
+const db = firebase.database();
+// Links to head of database.
+const ref = db.ref("live_weller");
+// Links to markets list.
+const marketsRef = ref.child("markets");
 
 router.get('/', function(req, res, next) {
 	res.render('data');
 });
 
+router.post('/', function(req, res) {
+	if (req.body.type === "general") {
+		/* Levels chart */
+		let levZero = 0;
+		let levOne = 0;
+		let levTwo = 0;
+		let levThree = 0;
+
+		/* Store Type chart */
+		let small = 0;
+		let medium = 0;
+		let large = 0;
+		let convenience = 0;
+		
+		marketsRef.once('value', function(snapshot) {
+			snapshot.forEach(function(childSnapshot) {
+				const childData = childSnapshot.val();
+				const level = parseInt(childData.marketInfo.marketLevel);
+				const type = childData.marketInfo.storeType;
+
+				if (level == 0) {levZero++;}
```

Comment:
this could be a little simpler by having a `levels` array where the index corresponds to a level and the value is the number of markets at that index/level, so counting up the levels would like like
```javascript
let levels = [];
for each market snapshot {
    const marketLevel = parseInt(childData.marketInfo.marketLevel);
    levels[marketLevel]++;
}
```
and a populated `levels` would look like `[ 2, 5, 1, 2 ]` with whatever values but basically an array with 2 level 0s, 5 level 1s, 1 level 2, and 2 level 3s.